core/src/main/kotlin/org/evomaster/core/problem/rest/OpenApiAccess.kt

```diff
@@ -36,23 +40,33 @@ object OpenApiAccess {
                 ?: throw SutProblemException("Failed to parse OpenApi schema: " + parseResults.messages.joinToString("\n"))
     }
 
-    fun getOpenAPIFromURL(openApiUrl: String): OpenAPI {
+    fun getOpenAPIFromURL(openApiUrl: String, authentication : HttpWsAuthenticationInfo = HttpWsNoAuth() ): OpenAPI {
 
         //could be either JSON or YAML
        val data = if(openApiUrl.startsWith("http", true)){
-           readFromRemoteServer(openApiUrl)
+           readFromRemoteServer(openApiUrl, authentication)
        } else {
            readFromDisk(openApiUrl)
        }
 
+        // if the swagger could not be retrieved because of unauthorized access, return an empty OpanAPI
+        if (data == unauthorizedSchemaAccess) {
+            return OpenAPI()
+        }
+
         return getOpenApi(data)
     }
 
-    private fun readFromRemoteServer(openApiUrl: String) : String{
-        val response = connectToServer(openApiUrl, 10)
+    private fun readFromRemoteServer(openApiUrl: String, authentication : HttpWsAuthenticationInfo) : String{
+        val response = connectToServer(openApiUrl, authentication, 10)
 
         val body = response.readEntity(String::class.java)
 
+        // this is returned if the swagger could not be accessed due to unauthorized request.
+        if (response.toString().contains("Unauthorized") && response.toString().contains("401")) {
```

Comment:
should not rely on `Unauthorized` be there. also, token like `Unauthorized` and `401` could be in text message unrelated to HTTP status. 
rather check directly `response.status`. not only for `401`, but also `403` (just in case)

Reply:
Hi Andrea,

Thanks for your comment. I have fixed this one.